rush-plugins/rush-resolver-cache-plugin/src/afterInstallAsync.ts

```diff
@@ -0,0 +1,357 @@
+    out += BASE32[mask & (buffer << (5 - bits))];
+  }
+
+  return out;
+}
+
+// https://github.com/pnpm/pnpm/blob/f394cfccda7bc519ceee8c33fc9b68a0f4235532/packages/dependency-path/src/index.ts#L167-L189
+function depPathToFilename(depPath: string): string {
+  let filename: string = depPathToFilenameUnescaped(depPath).replace(/[\\/:*?"<>|]/g, '+');
+  if (filename.includes('(')) {
+    filename = filename.replace(/(\)\()|\(/g, '_').replace(/\)$/, '');
+  }
+  if (filename.length > 120 || (filename !== filename.toLowerCase() && !filename.startsWith('file+'))) {
+    return `${filename.substring(0, MAX_LENGTH_WITHOUT_HASH)}_${createBase32Hash(filename)}`;
+  }
+  return filename;
+}
+
+function depPathToFilenameUnescaped(depPath: string): string {
+  if (depPath.indexOf('file:') !== 0) {
+    if (depPath.startsWith('/')) {
+      depPath = depPath.slice(1);
+    }
+    return depPath;
+  }
+  return depPath.replace(':', '+');
+}
+
+interface IResolverContext {
+  descriptionFileRoot: string;
+  descriptionFileHash: string;
+  name: string;
+  deps: Map<string, string>;
+  isProject: boolean;
+  ordinal: number;
+  optional?: boolean;
+  files?: string[];
+}
+
+type IDependencyEntry =
+  | string
+  | {
+      version: string;
+      specifier: string;
+    };
+
+export async function afterInstallAsync(
+  rushSession: RushSession,
+  rushConfiguration: RushConfiguration,
+  subspace: Subspace,
+  logger: ILogger
+): Promise<void> {
+  const { terminal } = logger;
+  const rushRoot: string = `${rushConfiguration.rushJsonFolder}/`;
+
+  const lockFile: string = subspace.getCommittedShrinkwrapFilename();
+  const workspaceRoot: string = subspace.getSubspaceTempFolderPath();
+
+  /**
+   *
+   * @param key - The key of the dependency
+   * @param specifier - The specifier in the lockfile for the dependency
+   * @param context - The owning package
+   * @returns The identifier for the dependency
+   */
+  function resolveDependencyKey(key: string, specifier: string, context: IResolverContext): string {
+    if (specifier.startsWith('/')) {
+      return getDescriptionFileRootFromKey(specifier);
+    } else if (specifier.startsWith('link:')) {
+      if (context.isProject) {
+        return path.resolve(context.descriptionFileRoot, specifier.slice(5));
+      } else {
+        return path.resolve(workspaceRoot, specifier.slice(5));
+      }
+    } else if (specifier.startsWith('file:')) {
+      return getDescriptionFileRootFromKey(specifier, key);
+    } else {
+      return getDescriptionFileRootFromKey(`/${key}@${specifier}`);
+    }
+  }
+
+  function resolveDependencies(
+    collection: Record<string, IDependencyEntry>,
+    context: IResolverContext
+  ): void {
+    for (const [key, value] of Object.entries(collection)) {
+      const version: string = typeof value === 'string' ? value : value.version;
+      const resolved: string = resolveDependencyKey(key, version, context);
+      if (!resolved.endsWith('dependency-sets/null')) {
```

Comment:
Have we formalized the dependency-sets concept?

Reply:
I forgot this was here. I will look into a generic way to trim that out.

Reply:
Turns out we don't actually need it, leaving those in the map will just have a few pointers into the void if the concept is used. Doesn't break resolution.